### PR TITLE
support syslog remote module

### PR DESCRIPTION
**Pull request by anas-shami**
support system logging on onyx using hostname / IP and change these attributes:
- Override trap priority for such class 
- Change trap level
- Append new host 
- Change port number
- Add/Remove logging filter.

* New module/feature

### CODE REVIEW

File: lib/ansible/modules/network/onyx/onyx_syslog_remote.py [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        required: true
+        type: str
+    port:
+        description:
+          - Set remote server destination port for log messages
+        type: int
+    trap:
+        description:
+          - Minimum severity level for messages to this syslog server
+        choices: ['alert', 'crit', 'debug', 'emerg', 'err', 'info', 'none', 'notice', 'warning']
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_syslog_remote.py]: please sort according to severity

File: lib/ansible/modules/network/onyx/onyx_syslog_remote.py [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+            override_class:
+                description:
+                  - Specify a class whose log level to override
+                choices: ['mgmt-front', 'mgmt-back', 'mgmt-core', 'events', 'debug-module', 'sx-sdk', 'mlx-daemons', 'protocol-stack']
+                required: True
+                type: str
+            override_priority:
+                description:
+                  -Specify a priority whose log level to override
+                choices: ['alert', 'crit', 'debug', 'emerg', 'err', 'info', 'none', 'notice', 'warning']
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_syslog_remote.py]: again, please sort according to severity

File: lib/ansible/modules/network/onyx/onyx_syslog_remote.py [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+
+import re
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.onyx.onyx import show_cmd
+from ansible.module_utils.network.onyx.onyx import BaseOnyxModule
+
+
+class OnyxSyslogRemoteModule(BaseOnyxModule):
+    MAX_PORT = 65535
+    LEVELS = ['alert', 'crit', 'debug', 'emerg', 'err', 'info', 'none', 'notice', 'warning']
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_syslog_remote.py]: please sort

File: lib/ansible/modules/network/onyx/onyx_syslog_remote.py [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+
+    def load_current_config(self):
+        self._current_config = dict()
+        current_config = self.show_logging().split('\n')
+        for line in current_config:
+            line = line.strip()
+            match = self.LOGGING_HOST.match(line)
+            if match:
+                host = match.group(1)
+                self._current_config[host] = dict()
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_syslog_remote.py:191]: add continue after each match, as there is no need to check other matches

File: lib/ansible/modules/network/onyx/onyx_syslog_remote.py [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+                host_config.get('filter_str', None) != required_config['filter_str']
+            if required_config['enabled']:
+                if is_change or not host_config:
+                    self._commands.append('logging {0} filter {1} {2}'.format(host, required_config['filter'], required_config['filter_str']))
+
+            else:
+                if 'filter' in host_config:
+                    self._commands.append('no logging {0} filter'.format(host))
+
+        if len(required_config) == 2:
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_syslog_remote.py]: this is not a good test, and could cause bugs in the future

**Review by samerd — CHANGES_REQUESTED** [review]
Thanks @anas-shami for your contribution.
please fix the requested changes

File: lib/ansible/modules/network/onyx/onyx_syslog_remote.py (hunk 5 above)
Comment (anas-shami) [reply lib/ansible/modules/network/onyx/onyx_syslog_remote.py]: Here I must make sure that there is no other keys but [enabled, host] which are required, So you prefer to add a flag and in each check for other keys set the flag to False?

File: lib/ansible/modules/network/onyx/onyx_syslog_remote.py [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+author: "Anas Shami (@anass)"
+short_description: Configure remote syslog module
+description:
+  - This module provides declarative management of syslog
+    on Mellanox ONYX network devices.
+notes:
+options:
+    enabled:
+        description:
+          - Disable/Enable logging feature
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_syslog_remote.py]: disable logging to given remote host

**Comment by anas-shami** [comment]
@samerd  please review

File: lib/ansible/modules/network/onyx/onyx_syslog_remote.py [hunk 7]
```diff
@@ -37,7 +37,7 @@
     trap:
         description:
           - Minimum severity level for messages to this syslog server
-        choices: ['alert', 'crit', 'debug', 'emerg', 'err', 'info', 'none', 'notice', 'warning']
+        choices: ['none', 'info', 'notice', 'alert', 'debug', 'warning', 'err', 'emerg', 'crit']
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_syslog_remote.py]: 'debug' should be less than 'info'
please consult with onyx developers regarding the used enum

**Review by samerd — COMMENTED** [review]
just a small comment about the logging levels order

**Comment by anas-shami** [comment]
@samerd done thank you, please review

**Review by samerd — APPROVED** [review]
looks great
thanks @anas-shami

**Comment by samerd** [comment]
@NilashishC 
could you please review and approve this PR if it ok?

**Pull request merged**